core/src/main/java/org/springframework/security/access/expression/method/ExpressionBasedPreInvocationAdvice.java

```diff
@@ -81,6 +81,9 @@ else if (mi.getArguments().length == 1) {
 						"A PreFilter expression was set but the method argument type"
 								+ arg.getClass() + " is not filterable");
 			}
+		} else if (mi.getArguments().length > 1) {
+			throw new IllegalArgumentException(
+					"Unable to define the method argument for filtering. Specify the filter target.");
```

Comment:
I think the error message here should use a different word than "define" since that might imply something else. Could we instead do something like "decide", "determine", or "locate"?

```suggestion
					"Unable to determine the method argument for filtering. Specify the filter target.");
```